content/blog/creating-and-publishing-new-bsp/_index.md

```diff
@@ -0,0 +1,249 @@
+---
+title: "Creating and Publishing a New BSP: A Step-by-Step Guide for Developers"
+date: 2024-06-24
+showAuthor: false
+authors:
+    - "juraj-michalek"
+tags: ["Embedded Systems", "ESP32", "ESP32-C3", "Espressif", "BSP", "Contributing"]
+---
+
+## Introduction
+
+The ESP Board Support Package (ESP-BSP) is a powerful tool that simplifies the development of applications for various ESP32-based boards. If you're interested in creating and publishing a new BSP for a third-party board, this guide will walk you through the process, from setting up your development environment to submitting your BSP to the ESP Component Registry. For this guide, we'll use the ESP32-C3-DevKit-RUST-1 as our example board.
+
+## Prerequisites
+
+Before you start, ensure you have the following:
+
+- **ESP-IDF**: The official development framework for the ESP32, properly installed and sourced in your shell.
```

Comment:
This is kind of interesting question about the terminology. Community is using "ESP32", which is like people using word google, to search for something. Espressif SoCs is quite impractical, yet I understand that it's "correct".
Also in case of search we will have more hits with ESP32, thatn Espressif SoC.
What about rephrashing this thing to framework for Espressif SoC (ESP32, ESP32-C3, ESP32-P4...)